punishments.js

```diff
@@ -1150,6 +1205,24 @@ Punishments.checkName = function (user, userid, registered) {
 		punishment = Punishments.userids.get(user.locked);
 		if (!punishment) punishment = ['LOCK', user.locked, 0];
 	}
+	if (!battleban && Punishments.isBattleBanned(user)) {
```

Comment:
Punishments isn't TypeScripted, or else this would have been caught:

This is equivalent to `!battleban && battleban`, which is always `false`.

Reply:
And it's a good thing it's always `false`, because the next two lines are guaranteed to crash.

Reply:
I think this was copied from the lock command or maybe HH made the change, I don’t really remember writing this part, what was the crash jw

Reply:
`Punishments.isBattleBanned(user)` returns `Punishment | undefined`, which isn't a compatible type to pass to `Punishments.roomUserids.get()`, which expects a `string`.